app/assets/javascripts/models/beta/iteration.js

```diff
@@ -13,12 +14,27 @@ const getIterationForDate = (date, project) => {
     : iterationQuantity + 1;
 };
 
-export const getDateForIterationNumber = (iterationNumber, project) => {
-  return moment(new Date(project.startDate))
-    .startOf("isoWeek")
-    .add(iterationNumber, "weeks")
-    .format("ddd MMM Do Y");
-};
+const getNextIterationNumber = (sprints) => {
+  const lastSprint = last(sprints);
+
+  return lastSprint ? lastSprint.number + 1 : 1;
+}
+
+const getStartDate = (iterationNumber, project) => {
+  const momentDate = moment(project.startDate);
+
+  if (iterationNumber === 1) {
+    return moment(project.startDate);
+  }
+
+  const missingDaysFromFirstSprint = (momentDate.isoWeekday() - project.iterationStartDay) % 7;
```

Comment:
What do you think about extracting this logic to somewhere else to make it more explicit?